Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java

```diff
@@ -101,4 +112,19 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
                     MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
         }
     }
+
+    private void refreshBidderToken(final Context context) {
+        if (isComputingToken.compareAndSet(false, true)) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    final String token = Vungle.getAvailableBidTokens(10);
```

Comment:
I think we need to call it on the main thread first, and then if it's null (took more than 4 sec) return the in-memory value and update it on the background.
Also, the next getBiddingToken() call should ideally interrupt this thread to give priority to the new call.

Reply:
yeah I think we should try to send stale/old values only when we cannot get a fresh value (when sdk returns a null)